unit_tests/test_redis_client_base.py

```diff
@@ -0,0 +1,225 @@
+            self.redis_client.unregister(is_blocking=True)
+
+    @patch("d3a_api_client.redis_client_base.wait_until_timeout_blocking",
+           side_effect=AssertionError)
+    def test_unregister_is_active_true_throws_exception(self,
+                                                        mock_wait_until_timeout_blocking,
+                                                        redis_client_auto_register):
+        """Checks if is active set to true throws exception."""
+        with pytest.raises(RedisAPIException,
+                           match="API unregister process timed out. "
+                                 "Server will continue processing "
+                                 "your request on the background and "
+                                 "will notify you as soon as "
+                                 "the unregistration has been completed."):
+            self.redis_client = redis_client_auto_register
+            self.redis_client.is_active = True
+            self.redis_client.unregister(is_blocking=True)
+
+    def test_on_register(self, redis_client_auto_register):
+        """Assigning the correct arguments to blocking command response and
+           checking by calling on_register function with correct message
+           and should not throw an exception."""
+        data = {"device_uuid": DEVICE_ID, "transaction_id": TRANSACTION_ID}
+        message = {"data": json.dumps(data)}
+        redis_client_auto_register._blocking_command_responses = {
+            "register": {"transaction_id": TRANSACTION_ID}}
+        redis_client_auto_register._on_register(message)
+
+    def test_on_unregister(self, redis_client_auto_register):
+        """Assigning the correct arguments to blocking command response and
+           checking by calling on_unregister function with correct message
+           and should not throw an exception."""
+        data = {"device_uuid": DEVICE_ID, "transaction_id": TRANSACTION_ID,
+                "response": "success"}
+        message = {"data": json.dumps(data)}
+        redis_client_auto_register._blocking_command_responses = {
+            "unregister": {"transaction_id": TRANSACTION_ID}}
+        redis_client_auto_register._on_unregister(message)
+
+    def test_on_unregister_throws_exception(self, redis_client_auto_register):
+        """Checks if response is not success should throw exception """
+        with pytest.raises(RedisAPIException,
+                           match=f"Failed to unregister from market {AREA_ID}."
+                                 "Deactivating connection."):
+            data = {"device_uuid": DEVICE_ID, "transaction_id": TRANSACTION_ID,
+                    "response": "unsuccessful"}
+            message = {"data": json.dumps(data)}
+            redis_client_auto_register._blocking_command_responses = {
+                "unregister": {"transaction_id": TRANSACTION_ID}}
+            redis_client_auto_register._on_unregister(message)
+
```

Comment:
Here also self.active should be checked.
```suggestion
assert redis_client_auto_register.active == True
```
However, I just saw that there is a bug in the code. `is_active` should only be changed if the command was successful IMO. What do you think?
![Screenshot 2021-08-31 at 09 47 54](https://user-images.githubusercontent.com/19326608/131463895-b0b9a819-e8dd-47cd-b5f1-e622abddc033.png)

Reply:
Yes, you are correct.

Reply:
@fievelk can you please add this change to the code `RedisClientBase._on_unregister` ? IMO this is a bug. Thanks!